src/Middleware/Rewrite/src/RedirectRule.cs

```diff
@@ -14,7 +15,7 @@ internal class RedirectRule : IRule
     public Regex InitialMatch { get; }
     public string Replacement { get; }
     public int StatusCode { get; }
-    public RedirectRule(string regex, string replacement, int statusCode)
+    public RedirectRule([StringSyntax(StringSyntaxAttribute.Regex)] string regex, string replacement, int statusCode)
     {
         if (string.IsNullOrEmpty(regex))
         {
```

Comment:
I can't comment on it, but as an aside, the `| RegexOptions.CultureInvariant` below is a nop.  CultureInvariant only means something when IgnoreCase is also specified.